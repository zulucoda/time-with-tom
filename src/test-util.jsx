import React from 'react';
import { oneOfType, shape, element, func } from 'prop-types';
import { Form, Formik } from 'formik';
import { BrowserRouter as Router } from 'react-router-dom';

/**
 * Formik Wrapper Test Util
 * @param {React.FunctionComponent} Component
 * @param {Object} props
 * @param {Object} initialValues
 * @param {Function} onSubmit
 * @returns {Formik}
 */
export const FormikWrapperTestUtil = ({
  Component,
  props = {},
  initialValues = {},
  onSubmit = jest.fn(),
}) => (
  <Formik initialValues={initialValues} onSubmit={onSubmit}>
    <Form>
      <Component {...props} />
    </Form>
  </Formik>
);

FormikWrapperTestUtil.propTypes = {
  Component: oneOfType([element.isRequired, func]),
  props: shape({}),
  initialValues: shape({}),
  onSubmit: func,
};

/**
 * React Router Wrapper Test Util
 * @param {React.FunctionComponent} Component
 * @param {object} props
 * @returns {React.FunctionComponent}
 */
export const ReactRouterWrapperTestUtil = ({ Component, props = {} }) => (
  <Router>
    <Component {...props} />
  </Router>
);

ReactRouterWrapperTestUtil.propTypes = {
  Component: oneOfType([element.isRequired, func]),
  props: shape({}),
};
