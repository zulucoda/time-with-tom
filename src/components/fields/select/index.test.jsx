import { render } from '@testing-library/react';
import { FormikWrapperTestUtil } from '../../../test-util';
import SelectField from '.';
import React from 'react';

describe('Components - Fields - Select - Unit Test', () => {
  test('does it render', () => {
    const props = {
      name: 'name',
      label: 'name',
      list: [],
    };
    const { container } = render(
      <FormikWrapperTestUtil Component={SelectField} props={props} />,
    );
    expect(container).toMatchSnapshot();
  });
  test('should render select with options', () => {
    const props = {
      name: 'name',
      label: 'name',
      list: [
        { title: 'some title 1', value: 'some value 1' },
        { title: 'some title 2', value: 'some value 2' },
        { title: 'some title 3', value: 'some value 3' },
        { title: 'some title 4', value: 'some value 4' },
        { title: 'some title 5', value: 'some value 5' },
        { title: 'some title 6', value: 'some value 6' },
        { title: 'some title 7', value: 'some value 7' },
        { title: 'some title 8', value: 'some value 8' },
        { title: 'some title 9', value: 'some value 9' },
      ],
    };
    const { container } = render(
      <FormikWrapperTestUtil Component={SelectField} props={props} />,
    );
    expect(container).toMatchSnapshot();
  });
});
