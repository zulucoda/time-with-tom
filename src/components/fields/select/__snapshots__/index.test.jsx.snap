// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Fields - Select - Unit Test does it render 1`] = `
<div>
  <form
    action="#"
  >
    <div>
      <label>
        name
      </label>
      <select
        name="name"
      />
    </div>
  </form>
</div>
`;

exports[`Components - Fields - Select - Unit Test should render select with options 1`] = `
<div>
  <form
    action="#"
  >
    <div>
      <label>
        name
      </label>
      <select
        name="name"
      >
        <option
          value="some value 1"
        >
          some title 1
        </option>
        <option
          value="some value 2"
        >
          some title 2
        </option>
        <option
          value="some value 3"
        >
          some title 3
        </option>
        <option
          value="some value 4"
        >
          some title 4
        </option>
        <option
          value="some value 5"
        >
          some title 5
        </option>
        <option
          value="some value 6"
        >
          some title 6
        </option>
        <option
          value="some value 7"
        >
          some title 7
        </option>
        <option
          value="some value 8"
        >
          some title 8
        </option>
        <option
          value="some value 9"
        >
          some title 9
        </option>
      </select>
    </div>
  </form>
</div>
`;
